lib/core/addFieldType.js

```diff
@@ -0,0 +1,16 @@
+// var debug = require('debug')('keystone:core:addCustomFieldType');
+
+// `typeSpec` must have `name` and `path` properties
+module.exports = function addFieldType (typeSpec) {
+	this.Field = this.Field || {};
+	this.Field.Types = this.Field.Types || {};
```

Comment:
Why do we need to be so defensive here? `keystone.Field` and `keystone.Field.Types` should always exist

Reply:
just a precaution, learned from hard lessons before Object destructing. In a perfect world, you wouldn't need this